src/metabase/notification/send.clj

```diff
@@ -57,48 +58,53 @@
+            retry-config    default-retry-config
+            retry-errors    (volatile! [])
+            retry-report    (fn []
+                              {:attempted_retries (count @retry-errors)
+                               ;; we want the last retry to be the most recent
+                               :retry_errors       (reverse @retry-errors)})
+            send!           (fn []
+                              (try
+                                (channel/send! channel message)
+                                (catch Exception e
+                                  (when (should-retry-sending? e (:type channel))
+                                    (vswap! retry-errors conj {:message   (u/strip-error e)
+                                                               :timestamp (t/offset-date-time)})
+                                    (log/warnf e "[Notification %d] Failed to send to channel %s , retrying..."
+                                               notification-id (handler->channel-name handler))
+                                    (throw e)))))
+            retrier         (retry/make retry-config)]
+        (log/debugf "[Notification %d] Sending a message to channel %s" notification-id (handler->channel-name handler))
+        (task-history/with-task-history {:task            "channel-send"
+                                         :on-success-info (fn [update-map _result]
+                                                            (cond-> update-map
+                                                              (seq @retry-errors)
+                                                              (update :task_details merge (retry-report))))
+                                         :on-fail-info    (fn [update-map _result]
+                                                            (update update-map :task_details merge (retry-report)))
+                                         :task_details    {:retry_config      retry-config
+                                                           :channel_id        (:id channel)
+                                                           :channel_type      (:type channel)
+                                                           :template_id       (:template_id handler)
+                                                           :notification_id   notification-id
+                                                           :notification_type payload-type
+                                                           :recipient_ids     (map :id (:recipients handler))}}
+
+          (retrier send!)
+          (log/debugf "[Notification %d] Sent to channel %s with %d retries"
+                      notification-id (handler->channel-name handler) (count @retry-errors))))
+      (prometheus/inc! :metabase-notification/channel-send-ok {:payload-type payload-type
+                                                               :channel-type channel-type} 1)
```

Comment:
nit: the 1 is implicit if you omit the last arg, and I prefer that stylically (less noise, more like clojure `inc`)